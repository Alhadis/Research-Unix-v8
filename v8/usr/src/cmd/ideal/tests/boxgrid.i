boxgrid
.nf
.IS
...minx -1.5
...maxy 1.5
...maxx 1.5
...miny -1.5
gridline {
	var a,b;
	var neg, pos;
	conn 0.5*(a+b) - neg to 0.5*(a+b) + pos;
}
rect {
	var ne, nw, sw, se;
	var ht, wd;
	var cen;
	se = sw + wd;
	ne = nw + wd;
	ne = se + (0,1)*ht;
	bdlist = ne, nw, sw, se;
	cen = 0.5*(ne+sw);
}
main{
	conn (0,-1) to (0,1) using 5 gridline {neg = 1; pos = 1;} <a,b>;
	conn (-1,0) to (1,0) using 5 gridline {neg = (0,1); pos = (0,1);} <a,b>;
	put rect{
		ht = wd = 0.5;
		cen = (0,-0.5);
		opaque;
	};
	put rect{
		ht = wd = 0.5;
		cen = (0,0.5);
		opaque;
	};
	put rect{
		ht = wd = 0.5;
		cen = 0.5;
		opaque;
	};
	put rect{
		ht = wd = 0.5;
		cen = -0.5;
		opaque;
	};
}
.IE
.IS
...forget rect gridline
.IE
