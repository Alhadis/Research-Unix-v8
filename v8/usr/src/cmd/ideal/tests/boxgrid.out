boxgrid
.nf
.IS
...minx -1.5
...maxy 1.5
...maxx 1.5
...miny -1.5
...maxx 1
...maxy 1
...minx -1
...miny -1
...line 0.8 1 0.8 -1
...line 0.4 1 0.4 0.25
...line 0.4 -0.25 0.4 -1
...line 0 1 0 0.75
...line 0 0.25 0 -0.25
...line 0 -0.75 0 -1
...line -0.4 1 -0.4 0.25
...line -0.4 -0.25 -0.4 -1
...line -0.8 1 -0.8 -1
...line 1 0.8 -1 0.8
...line 1 0.4 0.25 0.4
...line -0.25 0.4 -1 0.4
...line 1 0 0.75 0
...line 0.25 0 -0.25 0
...line -0.75 0 -1 0
...line 1 -0.4 0.25 -0.4
...line -0.25 -0.4 -1 -0.4
...line 1 -0.8 -1 -0.8
.IE
.IS
.IE
